src/Microsoft.AspNetCore.Sockets.Client/IConnection.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.AspNetCore.Sockets.Client
 {
     public interface IConnection
     {
-        Task StartAsync(ITransport transport, HttpClient httpClient);
+        Task StartAsync(ITransportFactory transportFactory, HttpClient httpClient);
```

Comment:
I started without http client and it did not look nice either because of HubConnection. The alternatives are - creating and tracking HttpClient in hubConnection when you use the overload that takes enum. Adding an overload of IConnection.StartAsync that takes transport type in addition to the one that takes factory and the HttpClient. Not having it as part of this interface also means that you basically always need to create your own HttpClient when using the transport factory and pass as the parameter to StartAsync instead of letting us create the HttpClient and give it to you. It seems that it basically always leaks somewhere.